crates/ra_hir/src/code_model_impl/module.rs

```diff
@@ -125,15 +130,37 @@ impl Module {
         );
 
         let segments = &path.segments;
-        for name in segments.iter() {
+        for (idx, name) in segments.iter().enumerate() {
             let curr = if let Some(r) = curr_per_ns.as_ref().take_types() {
                 r
             } else {
                 return Ok(PerNs::none());
             };
             let module = match curr.resolve(db)? {
                 Def::Module(it) => it,
-                Def::EnumVariant(ev) => return Ok(PerNs::values(ev.def_id)),
+                Def::Enum(e) => {
+                    if let Some(suffix) = segments.get(idx + 1) {
+                        assert!(segments.len() == idx);
+
+                        let matching_variant = e
+                            .variant_ids()
+                            .iter()
+                            .map(|&variant_id| {
+                                let variant = EnumVariant::new(variant_id, e.def_id());
+                                let variant_name: Name = variant
+                                    .name(db)
+                                    .expect("enum variant should have a name (1)")
```

Comment:
I think that this `expect(1)` is a cancellable. We should be able to use `?` here?

Reply:
Yup, that's way better. Thanks

Reply:
Can't seem to convert `Option`s to `Cancellable` tho :-(

Reply:
The first expect shouldn't be an `Option`? We should do `variant.name(db)?.expect("")` (possibly then removing the expect later if it could fail (e.g. due to an invalid AST))

Reply:
I've propagated the error now so that it doesn't panic at all on unexpected results